apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -1474,6 +1480,142 @@ impl FederatedQueryGraphBuilder {
         Ok(())
     }
 
+    fn handle_context(&mut self) -> Result<(), FederationError> {
+        let mut subgraph_to_args: MultiMap<Arc<str>, ObjectFieldArgumentDefinitionPosition> =
+            MultiMap::new();
+        let mut coordinate_map: HashMap<
+            Arc<str>,
+            HashMap<ObjectFieldDefinitionPosition, Vec<ContextCondition>>,
+        > = HashMap::default();
+        for (subgraph_name, subgraph) in self.base.query_graph.subgraphs() {
+            // TODO: Most other `handle_*` methods filter out
+            // `self.base.query_graph.source == subgraph_name`. Should we do something similar?
+            // Perhaps only process the subgraph with the name `self.base.query_graph.source`?
+            let Some((_, context_refs)) = &subgraph
+                .referencers()
+                .directives
+                .iter()
+                // TODO: I think I need to get the subgraph data which contains the name of the
+                // context in that subgraph, ya?
+                .find(|(dir, _)| **dir == CONTEXT_DIRECTIVE_NAME)
+            else {
+                continue;
+            };
+
+            let Some((_, from_context_refs)) = &subgraph
+                .referencers()
+                .directives
+                .iter()
+                // TODO: I think I need to get the subgraph data which contains the name of
+                // fromContext in that subgraph, ya?
+                .find(|(dir, _)| **dir == FROM_CONTEXT_DIRECTIVE_NAME)
+            else {
+                continue;
```

Comment:
One not about this: This is silencing potential errors. There could be `@context` in the subgraph but not `@fromContext` (or worse, visa versa). This simply ignore that. I'm comfortable with this because composition *should* catch that, but it would also be reasonable track this better and raise an error.